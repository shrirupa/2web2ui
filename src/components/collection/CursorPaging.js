import styles from './CursorPaging.module.scss';
import { FirstPage } from '@sparkpost/matchbox-icons';
import React from 'react';
import { Button, Pager } from '@sparkpost/matchbox';

export const CursorPaging = ({ currentPage, handlePageChange, previousDisabled, nextDisabled, handleFirstPage, perPage, totalCount }) => (
  //previousDisabled and nextDisabled are boolean values.

  <div className={styles.PageButtons}>
    <Button className = {styles.RewindButton}
      key='rewind'
      onClick={handleFirstPage}
    >
      <FirstPage/>
    </Button>
    <Pager>
      <Pager.Previous
        onClick={() => handlePageChange(currentPage - 1)}
        disabled={previousDisabled}/>
      <Pager.Next
        onClick={() => handlePageChange(currentPage + 1)}
        disabled = {nextDisabled}/>
    </Pager>
    <span className={styles.PageDisplay}
    >Page: <strong className={styles.Bold}>{currentPage}</strong> / {Math.ceil(totalCount / perPage)}</span>
  </div>
);
export default CursorPaging;
