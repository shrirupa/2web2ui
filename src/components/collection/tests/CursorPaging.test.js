import React from 'react';
import { shallow } from 'enzyme';
import CursorPaging from '../CursorPaging';

describe('CursorPaging', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      currentPage: 2,
      handlePageChange: jest.fn(),
      previousDisabled: false,
      nextDisabled: false,
      handleFirstPage: jest.fn(),
      perPage: 25,
      totalCount: 100
    };
    wrapper = shallow(<CursorPaging {...props} />);
  });

  it('should render cursor paging', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly render cursor paging when there is only one page', () => {
    const newProps = {
      currentPage: 1,
      handlePageChange: jest.fn(),
      previousDisabled: true,
      nextDisabled: true,
      handleFirstPage: jest.fn(),
      perPage: 25,
      totalCount: 20
    };
    wrapper = shallow(<CursorPaging {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle clicking rewind to first page', () => {
    wrapper.find('Button').first().simulate('click');
    expect(props.handleFirstPage).toHaveBeenCalled();
  });

  it('should handle clicking previous button', () => {
    wrapper.find('Pager').children().first().simulate('click');
    expect(props.handlePageChange).toHaveBeenCalledWith(props.currentPage - 1);
  });

  it('should handle clicking next button', () => {
    wrapper.find('Pager').children().last().simulate('click');
    expect(props.handlePageChange).toHaveBeenCalledWith(props.currentPage + 1);
  });
});
