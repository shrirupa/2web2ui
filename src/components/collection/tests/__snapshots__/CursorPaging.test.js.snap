// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CursorPaging should correctly render cursor paging when there is only one page 1`] = `
<div
  className="PageButtons"
>
  <Button
    className="RewindButton"
    key="rewind"
    onClick={[MockFunction]}
    size="default"
  >
    <FirstPage />
  </Button>
  <Pager>
    <Pager.Previous
      disabled={true}
      onClick={[Function]}
    />
    <Pager.Next
      disabled={true}
      onClick={[Function]}
    />
  </Pager>
  <span
    className="PageDisplay"
  >
    Page: 
    <strong
      className="Bold"
    >
      1
    </strong>
     / 
    1
  </span>
</div>
`;

exports[`CursorPaging should render cursor paging 1`] = `
<div
  className="PageButtons"
>
  <Button
    className="RewindButton"
    key="rewind"
    onClick={[MockFunction]}
    size="default"
  >
    <FirstPage />
  </Button>
  <Pager>
    <Pager.Previous
      disabled={false}
      onClick={[Function]}
    />
    <Pager.Next
      disabled={false}
      onClick={[Function]}
    />
  </Pager>
  <span
    className="PageDisplay"
  >
    Page: 
    <strong
      className="Bold"
    >
      2
    </strong>
     / 
    4
  </span>
</div>
`;
